src/Orleans/GrainDirectory/IGrainRegistrar.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans.GrainDirectory
+{
+    /// <summary>
+    /// A grain registrar takes responsibility of coordinating the registration of a grains,
+    /// possibly involving multiple clusters. 
+    /// The grain registrar is called only on the silo that is the owner for that grain.
+    /// </summary>
+    interface IGrainRegistrar
```

Comment:
They are not necessary for now - we just call them for each individual address. It is possible we may consider adding a batched implementation of the global-single-instance protocol at some point, if we find that performance suffers when registering/unregistering/deleting large numbers of addresses at a time. That is not an easy change however.